File: regexlib/python_re_test_file/regexlib_8003.py
# 8003
# (AUX|PRN|NUL|COM\d|LPT\d)+\s*$
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"AUX"*5000+"!_1SLQ_1"

import re
from time import perf_counter

regex = """(AUX|PRN|NUL|COM\d|LPT\d)+\s*$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "AUX" * i * 10000 + "!_1SLQ_1"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_8003.js
/* 8003
 * (AUX|PRN|NUL|COM\d|LPT\d)+\s*$
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"AUX"*5000+"!_1SLQ_1"
 */
var REGEX = /(AUX|PRN|NUL|COM\d|LPT\d)+\s*$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'AUX'.repeat(i*10000) + '!_1SLQ_1'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}